plugins/inputs/elasticsearch_query/elasticsearch_query.go

```diff
@@ -0,0 +1,277 @@
+package elasticsearch_query
+
+import (
+	"context"
+	"fmt"
+	"net/http"
+	"strconv"
+	"strings"
+	"sync"
+	"time"
+
+	elastic5 "gopkg.in/olivere/elastic.v5"
+
+	"github.com/influxdata/telegraf"
+	"github.com/influxdata/telegraf/internal"
+	"github.com/influxdata/telegraf/plugins/common/tls"
+	"github.com/influxdata/telegraf/plugins/inputs"
+)
+
+const sampleConfig = `
+  ## The full HTTP endpoint URL for your Elasticsearch instance
+  ## Multiple urls can be specified as part of the same cluster,
+  ## this means that only ONE of the urls will be written to each interval.
+  urls = [ "http://node1.es.example.com:9200" ] # required.
+
+  ## Elasticsearch client timeout, defaults to "5s".
+  # timeout = "5s"
+
+  ## Set to true to ask Elasticsearch a list of all cluster nodes,
+  ## thus it is not necessary to list all nodes in the urls config option
+  # enable_sniffer = false
+
+  ## Set the interval to check if the Elasticsearch nodes are available
+  ## This option is only used if enable_sniffer is also set (0s to disable it)
+  # health_check_interval = "10s"
+
+  ## HTTP basic authentication details (eg. when using x-pack)
+  # username = "telegraf"
+  # password = "mypassword"
+
+  ## Optional TLS Config
+  # tls_ca = "/etc/telegraf/ca.pem"
+  # tls_cert = "/etc/telegraf/cert.pem"
+  # tls_key = "/etc/telegraf/key.pem"
+  ## Use TLS but skip chain & host verification
+  # insecure_skip_verify = false
+
+[[inputs.elasticsearch_query.aggregation]]
+  ## measurement name for the results of the aggregation query
+  measurement_name = "measurement"
+
+  ## Elasticsearch indexes to query (accept wildcards).
+  index = "index-*"
+
+  ## The date/time field in the Elasticsearch index (mandatory).
+  date_field = "@timestamp"
+
+  ## Time window to query (eg. "1m" to query documents from last minute).
+  ## Normally should be set to same as collection interval
+  query_period = "1m"
+
+  ## Lucene query to filter results
+  filter_query = "*"
+
+  ## Fields to aggregate values (must be numeric fields)
+  metric_fields = ["metric"]
+
+  ## Aggregation function to use on the metric fields
+  ## Valid values are: avg, sum, min, max, sum
+  metric_function = "avg"
+
+  ## Fields to be used as tags
+  ## Must be non-analyzed fields, aggregations are performed per tag
+  tags = ["field.keyword", "field2.keyword"]
+
+  ## Set to true to not ignore documents when the tag(s) above are missing
+  # include_missing_tag = false
+
+  ## String value of the tag when the tag does not exist
+  ## Used when include_missing_tag is true
+  # missing_tag_value = "null"
```

Comment:
Are all of the non-commented options mandatory and are different to their defaults? If not, please comment as many of them as possible to give the user a sense of what is the bare-minimum to configure.